code/modules/mob/dead/observer/login.dm

```diff
@@ -3,6 +3,7 @@
 	if(!. || !client)
 		return FALSE
 
+	login_time = world.time
```

Comment:
not the right place for this.

login is called when the CLIENT gains control, not the ckey.

so if they disconnect and reconnect this will reset, and if they gain control while disconnected, this won't be called until they connect.

you should track this in the player details datum